airflow/contrib/hooks/gcp_dataflow_hook.py

```diff
@@ -216,7 +620,12 @@ def _set_variables(variables):
         return variables
 
     def start_java_dataflow(self, job_name, variables, dataflow, job_class=None,
-                            append_job_name=True):
+        append_job_name=True, cancel_existing_job=False, context=None):
+        self._validate_params(cancel_existing_job, append_job_name)
+        if not append_job_name and context:
+            # Add execution time to job name if we are not appending a random suffix
+            job_name = self._add_execution_date_to_job_name(job_name, context)
+
         name = self._build_dataflow_job_name(job_name, append_job_name)
```

Comment:
This would be executed even when the append job is false. Can we ignore invoking the method in that case?

PS: It might be that, it would not do anything when it is false, but not read friendly.

Reply:
I understand this but the "build_dataflow_job_name" fn also validates the job name (raises exception if its not valid) and normalizes it. So it needs to be present.

Reply:
okay.